server/index.js

```diff
@@ -108,6 +110,53 @@ app.get('/api/authorize', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.use(authorizationMiddleware);
+
+app.post('/api/search', (req, res, next) => {
+  const { keywords, subreddits, sendToInbox } = req.body;
+  if (!keywords || !subreddits || sendToInbox === null) {
+    throw new ClientError(400, 'missing search terms');
+  }
+
+  const sql = `
+    select *
+      from "users"
+     where "userId" = $1;
+  `;
+  const params = [req.user.userId];
+  db.query(sql, params)
+    .then(result => {
+      const [user] = result.rows;
+      if (!user) {
+        throw new ClientError(401, 'user not found');
+      }
```

Comment:
You should be able to move this check into your authorization middleware and then stick the user on `req.user`. That should simplify this route a bit.